app/javascript/packages/document-capture/components/in-person-full-address-search.spec.tsx

```diff
@@ -10,6 +10,92 @@ import { LOCATIONS_URL } from './in-person-location-post-office-search-step';
+          <FullAddressSearch onFoundLocations={handleLocationsFound} locationsURL={LOCATIONS_URL} />
+        </SWRConfig>,
+      );
+
+      await userEvent.click(
+        await findByText('in_person_proofing.body.location.po_search.search_button'),
+      );
+
+      const errors = await findAllByText('simple_form.required.text');
+      expect(errors).to.have.lengthOf(4);
+    });
+
+    it('displays an error for an invalid ZIP code length (length = 1)', async () => {
+      const handleLocationsFound = sandbox.stub();
+      const { findByText, findByLabelText, findAllByText } = render(
+        <SWRConfig value={{ provider: () => new Map() }}>
+          <FullAddressSearch onFoundLocations={handleLocationsFound} locationsURL={LOCATIONS_URL} />
+        </SWRConfig>,
+      );
+
+      await userEvent.type(
+        await findByLabelText('in_person_proofing.body.location.po_search.address_label'),
+        '200 main',
+      );
+      await userEvent.type(
+        await findByLabelText('in_person_proofing.body.location.po_search.city_label'),
+        'Endeavor',
+      );
+      await userEvent.type(
+        await findByLabelText('in_person_proofing.body.location.po_search.state_label'),
+        'DE',
+      );
+      await userEvent.type(
+        await findByLabelText('in_person_proofing.body.location.po_search.zipcode_label'),
+        '1',
+      );
+      await userEvent.click(
+        await findByText('in_person_proofing.body.location.po_search.search_button'),
+      );
+
+      const errors = await findAllByText('idv.errors.pattern_mismatch.zipcode_five');
+      expect(errors).to.have.lengthOf(1);
+    });
+
+    it('does not display an error for a valid ZIP code length (length = 5)', async () => {
+      const handleLocationsFound = sandbox.stub();
+      const { findByText, findByLabelText } = render(
+        <SWRConfig value={{ provider: () => new Map() }}>
+          <FullAddressSearch onFoundLocations={handleLocationsFound} locationsURL={LOCATIONS_URL} />
+        </SWRConfig>,
+      );
+
+      await userEvent.type(
+        await findByLabelText('in_person_proofing.body.location.po_search.address_label'),
+        '200 main',
+      );
+      await userEvent.type(
+        await findByLabelText('in_person_proofing.body.location.po_search.city_label'),
+        'Endeavor',
+      );
+      await userEvent.type(
+        await findByLabelText('in_person_proofing.body.location.po_search.state_label'),
+        'DE',
+      );
+      await userEvent.type(
+        await findByLabelText('in_person_proofing.body.location.po_search.zipcode_label'),
+        '17201',
+      );
+      await userEvent.click(
+        await findByText('in_person_proofing.body.location.po_search.search_button'),
+      );
+
+      try {
+        await findByText('idv.errors.pattern_mismatch.zipcode');
+      } catch (err) {
+        expect(err === 'Error: Unable to find an element with the text');
+      }
```

Comment:
This usage of `try`/`catch` means that the test will pass regardless of whether the error occurs. Generally it is best practice to avoid conditional behavior in tests.

I recommend something like this instead, i.e. using `queryByText` (synchronous, will not throw error).

```suggestion
      expect(queryByText('idv.errors.pattern_mismatch.zipcode')).to.be(null);
```

Reply:
I did not take your commit but something close, I modified to `expect(queryByText('idv.errors.pattern_mismatch.zipcode')).to.be.null;` as I was getting an error.